CMakeLists.txt

```diff
@@ -287,4 +292,11 @@ if (MSVC AND WIN32 AND NOT MSVC_VERSION VERSION_LESS 142)
     target_link_options(RtglExample PRIVATE $<$<CONFIG:Debug>:/INCREMENTAL>)
     target_compile_options(RtglExample PRIVATE $<$<CONFIG:Debug>:/ZI>)
 endif()
-endif()
\ No newline at end of file
+endif()
+
+# Build shaders as custom target
+add_custom_target(Shaders
+                  COMMAND python3 ${CMAKE_CURRENT_SOURCE_DIR}/Source/Shaders/GenerateShaders.py -builddir ${CMAKE_CURRENT_BINARY_DIR}
+                  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/Source/Shaders)
+
+add_dependencies(RayTracedGL1 Shaders)
```

Comment:
For compatibility with CI, and VS Code workflow, can we add an option:

* `option(RG_WITH_SHADERS "Build includes a shader generation" ON)` ?

Then we can specify `-DRG_WITH_SHADERS=OFF` in `.github/workflows/build-windows.yml` for compatibility with CI